test/remote-method.test.js

```diff
@@ -0,0 +1,76 @@
+/*global describe, beforeEach, it */
+'use strict';
+var path = require('path');
+var helpers = require('yeoman-generator').test;
+var wsModels = require('loopback-workspace').models;
+var SANDBOX =  path.resolve(__dirname, 'sandbox');
+var fs = require('fs');
+var expect = require('chai').expect;
+var common = require('./common');
+
+describe('loopback:remote-method generator', function() {
+  beforeEach(common.resetWorkspace);
+  beforeEach(function createSandbox(done) {
+    helpers.testDirectory(SANDBOX, done);
+  });
+
+  beforeEach(function createProject(done) {
+    common.createDummyProject(SANDBOX, 'test-app', done);
+  });
+
+  beforeEach(function createCarModel(done) {
+    var test = this;
+    wsModels.ModelDefinition.create(
+      {
+        name: 'Car',
+        facetName: 'common'
+      },
+      function(err, model) {
+        test.Model = model;
+        done(err);
+      });
+  });
+
+  it('adds an entry to common/models/{name}.json', function(done) {
+    var methodGenerator = givenMethodGenerator();
+    helpers.mockPrompt(methodGenerator, {
+      model: 'Car',
+      methodName: 'myRemote',
+      isStatic: 'true',
+      desription: 'This is my first remote method',
+      httpPath: '/myRemote',
+      httpVerb: 'post',
+      acceptsArg: '',
+      returnsArg: ''
+    });
+
+    methodGenerator.run(function() {
+      var definition = readJsonSync('common/models/car.json');
+      var meths = definition.methods || {};
+      expect(meths).to.have.property('myRemote');
+      expect(meths.myRemote).to.eql({
+        isStatic: true,
+        accepts: [],
+        returns: [],
+        http: {
+          path: '/myRemote',
+          verb: 'post'
+        }
+      });
+      done();
+    });
+  });
+
+  function givenMethodGenerator() {
+    var name = 'loopback:remote-method';
+    var path = '../../remote-method';
+    var gen = common.createGenerator(name, path);
+    return gen;
+  }
+
+  function readJsonSync(relativePath) {
```

Comment:
This function is duplicated in several test files. Please extract it to `test/common.js`, e.g. `common.readSandboxJsonSync(relativePath)`.

Ideally, this refactoring should be a standalone commit, the first one if this patch (use `git rebase -i master && git push -f`). Alternatively, you can send this refactoring in a new pull request that we can quickly land so that you can rebase this PR on top of that change. Please let me know what would be the easiest way for your.